### DISCUSSION TITLE
Working in Huggingface

### DISCUSSION THREAD

**Discussion by evanbrociner** [zhudotexe/kani#21] [Q&A] [answered]
Hi thank you for this great repo. Any chance you can provide any example of the Huggingface engine. I am having trouble getting it to work. Thank you!

**Comment by zhudotexe** [comment]
Sure - if you're working with LLaMA v2 or Vicuna (or models that use the same prompt format) we have engines implemented for each of these already and you can just pass `model_id`. If you are looking to implement your own chat model on HuggingFace, you can use the Vicuna engine as an example of how to implement `build_prompt` and `message_len`: https://github.com/zhudotexe/kani/blob/main/kani/engines/huggingface/vicuna.py

See also https://kani.readthedocs.io/en/latest/engines.html#huggingface for more information on the HuggingEngine.

**Comment by zhudotexe** [comment, marked as answer]
Moving this issue to discussions.
